### ISSUE TITLE
Increase test coverage

### ISSUE THREAD

**Issue by JPDSousa** [JPDSousa/mongo-obj-framework#25]
Hacktoberfester? :octocat:
------------
Hey Hacktoberfester, if you came across this issue and you like MongoDB (and java), you've just found an awesome project to contribute to!

If you're not so much into databases and ORMs but you would like to contribute to other projects of mine: [search my other Hacktoberfest issues](https://github.com/search?utf8=%E2%9C%93&q=label%3Ahacktoberfest+state%3Aopen+type%3Aissue+author%3AJPDSousa&type=)

Description :mag:
----------
Test coverage is really useful in any kind of software project. Smof greatly appreciates test coverage contributions.

How to? :anguished:
------
:warning: Do not panic! Test coverage is not rocket science! :warning:
Smof uses [Maven](https://maven.apache.org/) to run tests. So, in order to run existing tests and create your own:
 - Clone the repository;
 - Be sure to have maven installed in your machine (check the [Maven](https://maven.apache.org/) homepage on how to download and install maven;
 - Start a MongoDB instance:
   - `mongod --port 20720`
   - You can, optionally, specify a path to the database or any other valid options, just make sure the port is set to 27020, which is the port used by Smof to run tests;
 - hit `mvn clean verify` in the projects root folder (i.e. mongo-obj-framework) by default, and wait for the green text stating that everything went fine;
 - Now you're good to go. Add tests to the test source folder `src/test/java` and increase the test coverage!
    - Remember to create test classes that end with `Test`, such as `ExampleTest.java`, as Maven Surefire plugin will only run classes with that pattern;
    - Focus on code that is not tested yet. You can resort to the codecov reports, or even to the reports generated by Maven Jacoco Plugin, stored in `{project.root}/target/site/jacoco` (open the index.html for a sort-of-pretty report).

Questions?
-------
Smof is in active development, and has not been properly tested, which may lead to some occasional bugs or questions. Feel free to create an [issue](https://github.com/JPDSousa/mongo-obj-framework/issues/new) if you stumble upon any bug or comment bellow if you have any douts.

**Comment by cunhazera** [comment]
@JPDSousa can you describe how to build the project and run it?
I tried to build it with mvn clean install, but did not work

**Comment by JPDSousa** [comment]
@cunhazera  The `mvn clean install`  command will run the tests, which require an active mongodb connection in port 27020. If you have mongodb installed, simply type `mongod --port 27020` in one console (this will create a mongodb server), and run `mvn clean install` in a separate terminal.

**Comment by archie-swif** [comment]
Hi folks!
I'm planning to add some tests for Or/AndQuery.

**Comment by JPDSousa** [comment]
@archie-swif That's really cool! Actually, the whole And/Or system might be a bit fragile ... i implemented it long time ago an never got the time to properly test it. But i totally encourage you to do it! I think it is a great feature (and totally useful for updates).

If you have any doubt or suggestion just share it! (might need to open a new issue for that)

**Comment by tminussi** [comment]
@JPDSousa thanks for merging ec31910. Step by step and we'll get a high coverage!

Count on me.

**Comment by boaglio** [comment]
@JPDSousa  is anything left to test ?

**Comment by JPDSousa** [comment]
@boaglio  there is still some basic code in the collection's package, and in the gridfs package. I'm currently working in a new cool feature which will bring some new code to the project as well...

**Comment by JPDSousa** [comment]
@tminussi I've made some upgrades to the parser package (#32), and decided to warn you in case your testing any parser code. The new logic i'm using is still incomplete (but functional) and it's still not fully tested, but i'll  add some base tests as soon as i can!

**Comment by tminussi** [comment]
@JPDSousa awesome. Thanks for letting me know!

**Comment by Nipel-Crumple** [comment]
@JPDSousa hi, I'd like to contribute too. Can I just take some untested piece of code, write tests and submit pr?

**Comment by JPDSousa** [comment]
@Nipel-Crumple sure! If you have any doubt just let me know!

**Comment by Nipel-Crumple** [comment]
@JPDSousa I'd want to test `SmofUpdate` class, and think about how to create test method for `org.smof.collection.SmofUpdate#rename`? Maybe you have some ideas? Cause if I rename field, the mapping is broken and field is null after `find()` from MongoDb

**Comment by JPDSousa** [comment]
You're right @Nipel-Crumple . Well, to be fair, i didn't think much about it when i added the `org.smof.collection.SmofUpdate#rename` method, as it is a bit useless to the project.

Maybe leave it untested for now. I'm going to tag it as `@Deprecated` (or even remove) in a future patch, while i think of a cool application for the MongoDB's renaming feature (useful for the project). In the mean time, if you have some ideas feel free to share 😄

**Comment by JPDSousa** [comment]
For anyone who is looking for code to test: #33 has brought a whole new set of querying features, some of them which are not tested yet. Feel free to extend [ReadOpTest](https://github.com/JPDSousa/mongo-obj-framework/blob/master/test/org/smof/ReadOpTest.java) with some new cool query tests.

**Comment by Nipel-Crumple** [comment]
Okay, I got it, thanks a lot 🙂  So I'll try write tests for other methods of this class

**Comment by lavanyajain** [comment]
[JPDSousa](https://github.com/JPDSousa) Any class that I can pickup here? if so please add jira and assign it to me

**Comment by JPDSousa** [comment]
Hey @lavanyajain this project is not active. Hence, we're not promoting future developments.
